src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1006 @@
+            tensors for more detail.
+
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+class BridgeTowerLayerNorm(nn.LayerNorm):
+    """Subclass torch's LayerNorm to handle fp16."""
+
+    def forward(self, x: torch.Tensor):
+        orig_type = x.dtype
+        ret = super().forward(x.type(torch.float32))
+        return ret.type(orig_type)
+
+
+class BridgeTowerResidualAttention(nn.Module):
+    def __init__(self, d_model: int, n_head: int, attn_mask: torch.Tensor = None):
+        super().__init__()
+
+        self.attn = nn.MultiheadAttention(d_model, n_head)
+        self.ln_1 = BridgeTowerLayerNorm(d_model)
+        self.mlp = nn.Sequential(
+            OrderedDict(
+                [
+                    ("c_fc", nn.Linear(d_model, d_model * 4)),
+                    ("gelu", QuickGELUActivation()),
+                    ("c_proj", nn.Linear(d_model * 4, d_model)),
+                ]
+            )
+        )
+        self.ln_2 = BridgeTowerLayerNorm(d_model)
+        self.attn_mask = attn_mask
+
+    def attention(self, x: torch.Tensor, x_mask: torch.Tensor):
+        if x_mask is not None:
+            x_mask = x_mask.to(dtype=torch.bool, device=x.device)
+        self.attn_mask = self.attn_mask.to(dtype=x.dtype, device=x.device) if self.attn_mask is not None else None
+        return self.attn(x, x, x, need_weights=False, attn_mask=self.attn_mask, key_padding_mask=x_mask)[0]
+
+    def forward(self, x: torch.Tensor, x_mask: torch.Tensor = None):
+        x = x + self.attention(self.ln_1(x), x_mask)
+        x = x + self.mlp(self.ln_2(x))
+        return x
+
+
+class BridgeTowerTransformer(nn.Module):
+    def __init__(
+        self,
+        width: int,
+        layers: int,
+        heads: int,
+        attn_mask: torch.Tensor = None,
+        model_type: str = "bridgetower",
+        stop_gradient: bool = False,
+        vit_remove_last: bool = False,
+    ):
+        super().__init__()
+        self.width = width
+        self.layers = layers
+        if vit_remove_last:
+            self.resblocks = nn.Sequential(
+                *[BridgeTowerResidualAttention(width, heads, attn_mask) for _ in range(layers - 1)]
+            )
+        else:
+            self.resblocks = nn.Sequential(
+                *[BridgeTowerResidualAttention(width, heads, attn_mask) for _ in range(layers)]
+            )
+        self.model_type = model_type
+        self.stop_gradient = stop_gradient
+
+    def forward(self, x: torch.Tensor, x_mask: torch.Tensor = None):
+        xs = []
+        for block in self.resblocks:
+            x = block(x, x_mask)
+            if self.model_type == "bridgetower":
```

Comment:
Why have this logic for `model_type`? As far as I can tell, it's always `"bridgetower"` as `BridgeTowerCLIP` always is set with the default.

Reply:
We'd like to retain model_type incase of any variants we'd potentially want to support in coming releases/PRs. Hence keeping it.

Reply:
A new `model_type` would require a new model folder. Transformers is not a modular toolbox and each model has their own modeling file. As a result, please remove this test for better readability.